src/Adapters/AzureManagementAdapter.ts

```diff
@@ -505,25 +480,27 @@ export default class AzureManagementAdapter implements IAzureManagementAdapter {
                         '[START] Applying search params (filter and take)'
                     );
                     // start with filter string to test the passed filter string against all possible display fields/properties in resource
-                    if (searchParams?.filter) {
+                    if (params.getResourcesParams.searchParams?.filter) {
                         resources = resources.filter((resource) => {
                             return Object.keys(AzureResourceDisplayFields)
                                 .filter((f) => isNaN(Number(f)))
                                 .some((displayField) =>
                                     resource[displayField]?.includes(
-                                        searchParams?.filter
+                                        params.getResourcesParams.searchParams
+                                            ?.filter
                                     )
                                 );
                         });
                     }
 
-                    // take the first n number of resources to make sure the browser won't crash with making thousands of requests
+                    // if necessary, take the first n number of resources to make sure the browser won't crash with making thousands of requests
                     // to check permissions for each resource, however, this might cause 0 result after checking the permissions for the first taken n resources
                     // it is hard to limit the number of requests being made and make sure to capture the resources that we might have required access permissions
-                    resources = resources.slice(
-                        0,
-                        searchParams?.take || MAX_RESOURCE_TAKE_LIMIT
-                    );
+                    // resources = resources.slice(
```

Comment:
wasn't this important? I thought you were seeing it crash your machine without this.

Reply:
Since now we are using storageContainerAccountId parameter to pass to the container picker it is not crashing and we don't need to slice the resources result. But I wanted to keep it as comments for the future if any of the other resource pickers crashes, users can try to slice the resources before making another batch of calls for role assignments.